numba/np/arrayobj.py

```diff
@@ -1658,25 +1684,8 @@ def fancy_setslice(context, builder, sig, args, index_types, indices):
             msg = "cannot assign slice from input of different size"
             context.call_conv.return_user_exc(builder, ValueError, (msg,))
 
-        # Check for array overlap
-        src_start, src_end = get_array_memory_extents(context, builder, srcty,
```

Comment:
Given this check is removed, does the proposed implementation correctly handle "overlap"?

Reply:
Define overlap in context of arrays ?

Reply:
With the current implementation, I don't think that's possible since fancy indexing always creates a copy of the data. i.e. In fancy indexing the source and destination array never share memory. This is stated in the NumPy documentation as well. 

I'm in-fact curious as of why it was the case over here that they did.